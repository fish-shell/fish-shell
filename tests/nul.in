# Check that splitting on NUL works
set -l var (printf '%s\0' "one line" two\nlines banana)
string escape -- $var
echo $var[2]

echo

# This is split into one element - keeping the trailing newline.
set -l var (printf '%s\n' "one line" two\nlines banana; printf '\0')
string escape -- $var

echo

# This is still split on newlines
set -l var (printf '%s\n' "one line" two\nlines banana)
string escape -- $var
