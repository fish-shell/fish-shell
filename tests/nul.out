'one line'
two\nlines
banana
two
lines

one\ line\ntwo\nlines\nbanana\n

'one line'
two
lines
banana
